acceptance/tests/vault/helpers.go

```diff
@@ -111,88 +106,49 @@ func configureEnterpriseLicenseVaultSecret(t *testing.T, vaultClient *vapi.Clien
 }
 
-// configureKubernetesAuthRoles configures roles for the Kubernetes auth method
+// configureKubernetesAuthRole configures a role for the component for the Kubernetes auth method
 // that will be used by the test Helm chart installation.
-func configureKubernetesAuthRoles(
-	t *testing.T,
-	vaultClient *vapi.Client,
-	consulReleaseName, ns, authPath, datacenter string,
-	cfg *config.TestConfig,
-	isPrimary bool,
-) {
-	consulClientServiceAccountName := fmt.Sprintf("%s-consul-client", consulReleaseName)
-	consulServerServiceAccountName := fmt.Sprintf("%s-consul-server", consulReleaseName)
-	sharedPolicies := "consul-gossip"
-	if cfg.EnableEnterprise {
-		sharedPolicies += ",consul-enterpriselicense"
-	}
+func configureKubernetesAuthRole(t *testing.T, vaultClient *vapi.Client, consulReleaseName, ns, authPath, component, policies string) {
+	componentServiceAccountName := fmt.Sprintf("%s-consul-%s", consulReleaseName, component)
 
 	// Create the Auth Roles for consul-server and consul-client.
 	// Auth roles bind policies to Kubernetes service accounts, which
 	// then enables the Vault agent init container to call 'vault login'
 	// with the Kubernetes auth method to obtain a Vault token.
 	// Please see https://www.vaultproject.io/docs/auth/kubernetes#configuration
 	// for more details.
-	logger.Log(t, "Creating the consul-server and consul-client roles")
+	logger.Logf(t, "Creating the %q", componentServiceAccountName)
 	params := map[string]interface{}{
-		"bound_service_account_names":      consulClientServiceAccountName,
-		"bound_service_account_namespaces": ns,
-		"policies":                         sharedPolicies,
-		"ttl":                              "24h",
-	}
-	_, err := vaultClient.Logical().Write(fmt.Sprintf("auth/%s/role/consul-client", authPath), params)
-	require.NoError(t, err)
-
-	// Both primary and secondary datacenters need access to the replication token, but
-	// only the primary needs to be able to read the bootstrap token.
-	policies := fmt.Sprintf(sharedPolicies+",connect-ca-%s,consul-server-%s,consul-replication-token", datacenter, datacenter)
-	if isPrimary {
-		policies += ",consul-bootstrap-token"
-	}
-	params = map[string]interface{}{
-		"bound_service_account_names":      consulServerServiceAccountName,
+		"bound_service_account_names":      componentServiceAccountName,
 		"bound_service_account_namespaces": ns,
 		"policies":                         policies,
 		"ttl":                              "24h",
 	}
-	_, err = vaultClient.Logical().Write(fmt.Sprintf("auth/%s/role/consul-server", authPath), params)
+	_, err := vaultClient.Logical().Write(fmt.Sprintf("auth/%s/role/%s", authPath, component), params)
 	require.NoError(t, err)
+}
 
+// configureKubernetesAuthRole configures a role that allows all service accounts within the installation
+// namespace access to the Consul server CA.
+func configureConsulCAKubernetesAuthRole(t *testing.T, vaultClient *vapi.Client, ns, authPath string) {
```

Comment:
Instead of having one function that configures all roles, I've split it out to having a function that configures one role. This is because it's become hard to do it all in one function as some components needed different policy access depending on the test and so there was little benefit to having it all in one.

Also, having one function per role makes it easier to see what type of access each component needs depending on configuration (which follows principle of least privilege).